components/feature/addons/src/main/java/mozilla/components/feature/addons/update/SupportedAddonsChecker.kt

```diff
@@ -0,0 +1,204 @@
+     */
+    override fun registerForChecks() {
+        WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
+                CHECKER_UNIQUE_PERIODIC_WORK_NAME,
+                ExistingPeriodicWorkPolicy.KEEP,
+                createPeriodicWorkerRequest()
+        )
+        logger.info("register for checking for new supported Addons")
+    }
+
+    /**
+     * See [SupportedAddonsChecker.unregisterForChecks]
+     */
+    override fun unregisterForChecks() {
+        WorkManager.getInstance(applicationContext)
+                .cancelUniqueWork(CHECKER_UNIQUE_PERIODIC_WORK_NAME)
+        logger.info("unregister for checking for new supported Addons")
+    }
+
+    @VisibleForTesting
+    internal fun createPeriodicWorkerRequest(): PeriodicWorkRequest {
+        val constraints = getWorkerConstrains()
+
+        return PeriodicWorkRequestBuilder<SupportedAddonsWorker>(
+                frequency.repeatInterval,
+                frequency.repeatIntervalTimeUnit
+        )
+                .setConstraints(constraints)
+                .addTag(WORK_TAG_PERIODIC)
+                .build()
+    }
+
+    @VisibleForTesting
+    internal fun getWorkerConstrains() = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+
+    companion object {
+        private const val IDENTIFIER_PREFIX = "mozilla.components.feature.addons.update"
+
+        @VisibleForTesting
+        internal const val CHECKER_UNIQUE_PERIODIC_WORK_NAME =
+                "$IDENTIFIER_PREFIX.DefaultSupportedAddonsChecker.periodicWork"
+
+        /**
+         * Identifies all the workers that periodically check for new add-ons.
+         */
+        @VisibleForTesting
+        internal const val WORK_TAG_PERIODIC =
+                "mozilla.components.feature.addons.update.DefaultSupportedAddonsChecker.periodicWork"
+
+        internal fun createDefaultNotificationIntent(content: Context): Intent {
+            return content.packageManager.getLaunchIntentForPackage(content.packageName)?.apply {
+                flags = Intent.FLAG_ACTIVITY_NEW_TASK
+            } ?: throw IllegalStateException("Package has no launcher intent")
+        }
+    }
+}
+
+/**
+ * A implementation which uses WorkManager APIs to perform check for new available supported add-ons.
+ */
+internal class SupportedAddonsWorker(
+    val context: Context,
+    params: WorkerParameters
+) : CoroutineWorker(context, params) {
+    private val logger = Logger("SupportedAddonsWorker")
+
+    @Suppress("TooGenericExceptionCaught", "MaxLineLength")
+    override suspend fun doWork(): Result {
+
+        return try {
+            logger.info("Trying to check for new supported add-ons")
+
+            val addonManager = GlobalAddonDependencyProvider.requireAddonManager()
+
+            val newSupportedAddons = addonManager.getAddons().filter { addon ->
+                !addon.isEnabled() && addon.isDisabledAsUnsupported()
```

Comment:
This needs to be `addon.isSupported() && addon.isDisabledAsUnsupported()`. Bascially the addon is now `supported = true` because AMO returns it in the collection, but it's marked by as "disabled as unsupported" by migration. I don't think we need the additional `isEnabled` check.